src/Framework/FieldsAPI/FieldCollection/Contract/GroupNode.php

```diff
@@ -4,4 +4,8 @@
 
 interface GroupNode extends Node {
 	public function getFields();
+	public function append( Node $node );
+	public function getNodeIndexByName( $name );
+	public function getNodeByName( $name );
+	public function count();
```

Comment:
Here’s the code that inspired this change:

https://github.com/impress-org/givewp/blob/d0176e99b0cc46a5f65d9d62eb905aae5d2ed483/src/Framework/FieldsAPI/FieldCollection.php#L53-L55

Before, instances of `GroupNode`s which did not have a `getNodeByName` method would break here. The quickfix would have just been to change `GroupNode` to `FieldCollection` in the condition, however, I thought it would be more beneficial to add the method in question as well as a few other desirable methods to the `GroupNode` contract.

Reply:
@knowler Yeah, I think this makes sense and was the original intent that `GroupNode`s be recursively searchable. Updating the check to `FieldCollection` would have been incorrect - good call.